airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -399,15 +403,14 @@ private AutoCloseableIterator<AirbyteMessage> createReadIterator(final Database
       final JsonSchemaPrimitive cursorType = IncrementalUtils.getCursorType(airbyteStream,
           cursorField);
 
+      CursorStateMessageProducer messageProducer = new CursorStateMessageProducer(
+          stateManager,
+          cursorInfo.map(CursorInfo::getCursor).orElse(null));
+
       iterator = AutoCloseableIterators.transform(
-          autoCloseableIterator -> new StateDecoratingIterator(
```

Comment:
Can we delete the `StateDecoratingIterator` class in this PR?

Reply:
Realized this is still referred by DynamodbSource.java. Will remove in a separate PR.

Reply:
SGTM, but let's tag that class as deprecated, at least.